### PR TITLE
Make script resources use properties

### PR DESCRIPTION
**Pull request by jkeiser**

### CODE REVIEW

**Comment by lamont-granquist** [comment]
pingles again.

**Comment by lamont-granquist** [comment]
looks like this one probably falls to me, i'll have to finish for merge it.

File: lib/chef/resource/script.rb [hunk 1]
```diff
@@ -27,26 +27,22 @@ class Resource
...
-      # FIXME: remove this and use an execute sub-resource instead of inheriting from Execute
-      def command(arg = nil)
-        unless arg.nil?
-          raise Chef::Exceptions::Script, "Do not use the command attribute on a #{resource_name} resource, use the 'code' attribute instead."
-        end
-        super
-      end
+      property :command, coerce: proc { |v|
+        # Chef-13: change this to raise if the user is trying to set a value here
+        Chef::Log.warn "Specifying command attribute on a script resource is a coding error, use the 'code' attribute, or the execute resource"
```
Comment (tas50) [inline lib/chef/resource/script.rb]: We're going to want to go back to failing if command is specified.

File: lib/chef/resource/script.rb [hunk 2]
```diff
@@ -27,26 +27,22 @@ class Resource
...
+        Chef::Log.warn "This attribute is deprecated and must be fixed or this code will fail on Chef 13"
+        v
+      }
 
-      property :code, String, required: true
```
Comment (tas50) [inline lib/chef/resource/script.rb]: This should be required again

**Comment by lamont-granquist** [comment]
should wait on this one until i get back, and i've had enough coffee to look through the guard interpreter stuff.

**Comment by tas50** [comment]
@lamont-granquist this is probably going to need some rework. I was just getting it restarted against master to get that ball rolling

**Comment by tas50** [comment]
Closing due to the drift

**Pull request closed without merging**